src/store/CheckoutContext.js

```diff
@@ -1,15 +1,148 @@
-import {useReducer,createContext} from 'react';
+import {useReducer,createContext, useEffect,useContext,useState} from 'react';
+import axios from '../useAxios'
+import {useAuth} from './AuthContext'
+import { successToast, warningToast } from '../UI/Toast/Toast';
 
 export const CheckoutContext=createContext();
 
+export const useCheckout=()=>useContext(CheckoutContext)
+
 export const CheckoutContextProvider=({children})=>{
+    const [loading,setLoading]=useState(false)
+    const {token,userId}=useAuth()
+    
+    const config = {
+        headers: {
+            Authorization: "Bearer " + token
+        }
+    }
+
+    useEffect(()=>{
+        (
+            async ()=>{
+                if(token){
+                    const {data:{data}}=await axios.get(`/api/addresses/${userId}`,config)
+                    dispatch({
+                        type:"ADD_USER_ADDRESSES",
+                        payload:[...data]
+                    })
+                }
+            }
+        )()
+    },[token,userId])
+
+    useEffect(()=>{
+        (
+            async ()=>{
+                if(token){
+                    const {data:{data}}=await axios.get(`/api/payments/${userId}`,config)
+                    dispatch({
+                        type:"ADD_USER_PAYMENTS",
+                        payload:[...data]
+                    })
+                }
+            }
+        )()
+    },[token,userId])
```

Comment:
How about we put both GET req in one useEffect?

Reply:
Hey Danish its a better approach to keep 2 different useEffects to do 2 different tasks